src/v/cluster/feature_backend.cc

```diff
@@ -42,6 +46,12 @@ feature_backend::apply_update(model::record_batch b) {
             });
       });
 
+    auto batch_offset = b.base_offset();
+    co_await _feature_table.invoke_on_all(
+      [batch_offset](features::feature_table& t) {
+          t.set_applied_offset(batch_offset);
+      });
```

Comment:
I was under the impression `invoke_on_all()` was a bit expensive, putting work into queues and such. Could we put it up in the visit calls?

By no means a hot path so feel free to punt.

Reply:
actually this will only be called for feature batches

Reply:
I considered putting it into the above invoke_on_all statements, but it becomes a bit repetitive.  Since we only call this a couple times during upgrades, erring on the side of clarity.